x-pack/plugins/triggers_actions_ui/public/application/sections/rules_list/components/rules_list.tsx

```diff
@@ -228,6 +227,8 @@ export const RulesList = ({
   const [isCloningRule, setIsCloningRule] = useState<boolean>(false);
   const [isLoading, setIsLoading] = useState<boolean>(false);
 
+  const [localRefresh, setLocalRefresh] = useState<Date>(new Date());
```

Comment:
we were not able to re-use the refresh that we used before, we need it one specially for tags.

Reply:
As discussed, the `rulesList` doesn't have a local state for a refresh, so this refresh functionality only works if the parent component implements it. The rule management page that renders the rules list does not implement it. 

Adding a `localRefresh` allows the rules list to refresh its child components if the parent doesn't implement this. Because the tags component does its own fetching, we have 3 options:

1. Expose a callback function from the child to the parent, the parent will call it to refresh
2. Use the refresh prop
3. Lift the state up from the child to the parent

I went with option 2